features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx

```diff
@@ -405,18 +405,18 @@ function EndpointOverview(props) {
     const saveAdvanceConfig = (advanceConfig) => {
         const endpointConfigProperty =
             getEndpointTypeProperty(advanceConfigOptions.type, advanceConfigOptions.category);
-        const endpoints = epConfig[endpointConfigProperty];
-        if (Array.isArray(endpoints)) {
+        const selectedEndpoints = JSON.parse(JSON.stringify(epConfig[endpointConfigProperty]));
```

Comment:
Now we can use cloneDeep library as I have added it using [1] PR 

Example usage [2]

[1] - https://github.com/wso2/carbon-apimgt/pull/6863
[2] - https://github.com/dushansilva/carbon-apimgt/blob/98708b873ea67d64440164fe4a026060855cfbf4/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Resources/Resources.jsx#L422

Reply:
yeah @menakaj shall we use that function now?

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/6880/commits/75f21fb4a62c1df7aa8cc2da583ac8ddee06e065